### PR TITLE
Expose `goto_help` for GDExtension on `ScriptEditor`

### PR DESCRIPTION
**Pull request by Naros**
Exposes a helper method `goto_help` on the `ScriptEditor` that enables GDExtension and other tooling to request help to be displayed for a given help encoded descriptor string.

### CODE REVIEW

**Comment by AThousandShips** [comment]
I'd suggest exposing it in `ScriptEditor` instead as you can access that

**Comment by Naros** [comment]
Hi @AThousandShips, do you think there is any way this can be included in 4.3, given its simplicity?

**Comment by AThousandShips** [comment]
It depends, the 4.x is simply because all new features are assigned to 4.x until they're decided on, we haven't hit feature freeze yet, and this is small, so if it gets approved relatively soon it could be included in 4.3, but the fact that it's marked 4.x doesn't mean it can't possibly end up in 4.3 :)

**Review by dsnopek — APPROVED** [review]
Seems good to me!

The only potential roadblock is if the editor folks don't want to expose this method for some reason, but to me it seems like it'd be useful for both GDExtension and GDScript.

File: doc/classes/ScriptEditor.xml [hunk 1]
```diff
@@ -34,6 +34,13 @@
 				Returns an array with all [Script] objects which are currently open in editor.
 			</description>
 		</method>
+		<method name="goto_help">
+			<return type="void" />
+			<param index="0" name="topic" type="String" />
+			<description>
+				Opens help for the given topic.
```
Comment (akien-mga) [inline doc/classes/ScriptEditor.xml:41]: It would be good to add details on what the topic is meant to be, with some examples.

Comment (Naros) [reply doc/classes/ScriptEditor.xml:41]: Hi @akien-mga, that's reasonable.  Just one question, I was considering appending this:
```
For example: [code]class_name:<class_name>[/code] or [code]class_method:<class_name>:<method_name>[/code].
```
My concern in this context are the `<` and `>` and how they should be represented.  I also tried to find some other examples in the repository and it wasn't immediately obvious of any I could use as a basis for how best to represent this type of example where there is expected tokens combined with variable user-supplied data.  If there is, could you point me to a reference?

Comment (Naros) [reply doc/classes/ScriptEditor.xml:41]: So the closest I found was in `EditorCommandPalette` that uses:
```
[code]"example/command1"[/code] then [code]example[/code] will be the section name.
```
If that's acceptable here in this context, I could use `"class"` and `"method"` to represent the user-supplied values to distinguish it from the expected static values that the help system expects.  wdyt?

Comment (Naros) [reply doc/classes/ScriptEditor.xml]: Updated, let me know if it needs any further adjustments.

Comment (akien-mga) [reply doc/classes/ScriptEditor.xml]: That looks ok, but there seems to be more types of topics supported by this method, see:

https://github.com/godotengine/godot/blob/7e4c150573d6af7072f2b55ae76dc7b723f21f66/editor/editor_help.cpp#L2246-L2306

So it should likely be a list or supported tags with their expected arguments.

And then a few real life examples like `class_method:Node:add_child`.

Would also be good to figure out what's the format used for GDScript class docs so that this could be shown as an example too (maybe it's just the registered `class_name`, I didn't check).

Comment (Naros) [reply doc/classes/ScriptEditor.xml]: So this is what I have come up with @AThousandShips / @akien-mga.  It's a bit wordy, so I'm certainly open to suggestions, but I believe it conveys the details you requested.

image

Comment (akien-mga) [reply doc/classes/ScriptEditor.xml]: That looks really good!

**Review by akien-mga — APPROVED** [review]
Looks great to me!

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by Naros** [comment]
> Thanks! And congrats for your first merged Godot contribution 🎉

Thanks :tada:
